pandas/tests/arrays/string_/test_string.py

```diff
@@ -478,25 +478,35 @@ def test_fillna_args(dtype, request, arrow_string_storage):
 def test_arrow_array(dtype):
     # protocol added in 0.15.0
     pa = pytest.importorskip("pyarrow")
+    import pyarrow.compute as pc
 
     data = pd.array(["a", "b", "c"], dtype=dtype)
     arr = pa.array(data)
     expected = pa.array(list(data), type=pa.string(), from_pandas=True)
     if dtype.storage in ("pyarrow", "pyarrow_numpy") and pa_version_under12p0:
         expected = pa.chunked_array(expected)
-
+    if dtype.storage == "pyarrow_numpy":
+        expected = pc.cast(arr, pa.large_string())
     assert arr.equals(expected)
 
 
 @pytest.mark.filterwarnings("ignore:Passing a BlockManager:DeprecationWarning")
-def test_arrow_roundtrip(dtype, string_storage2):
+def test_arrow_roundtrip(dtype, string_storage2, request):
     # roundtrip possible from arrow 1.0.0
     pa = pytest.importorskip("pyarrow")
 
+    if dtype.storage == "pyarrow_numpy" and string_storage2 == "pyarrow":
```

Comment:
IMO we should just change both to be large_string

Reply:
Fine by me but not sure if we should deprecate the other one first?

Reply:
Is there any behavior difference expected between string and large string? If not, I don't think this needs a deprecration. I would consider it an implementation detail / feature

Reply:
I would also change both at the same time (officially String dtype is also still considered as experimental). 

It will change your schema when you convert to Arrow, and so for sure people will have things to update, although I assume (hope) it will be mostly tests that are checking the exact type.

Reply:
Makes sense. Guessing some things at the binary level (ex: pickle compatibility) might change across versions too

Reply:
I'll make the change later and then we can merge

I can monitor some of the low level stuff on the Dask CI

Reply:
Updated to switch to large strings for both